src/utilities/checkDynamicOptionSupport.ts

```diff
@@ -0,0 +1,68 @@
+"use strict";
+
+import * as fs from "fs-extra";
+
+import { recursivelyListAllFilePathsForTemplatePath } from "./getTemplateFilesAndFolders";
+
+export async function templatePathContainsDynamicOptions(templatePath: string): Promise<boolean> {
+  const fileList = await recursivelyListAllFilePathsForTemplatePath(templatePath);
+
+  for (const filePath of fileList) {
+    const result = await findDynamicOptionTokenInFile(filePath);
+    if (result) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+export async function getTemplatePathDynamicOptions(templatePath: string): Promise<string[]> {
+  const fileList = await recursivelyListAllFilePathsForTemplatePath(templatePath);
+
+  let result: string[] = [];
+
+  for (const filePath of fileList) {
+    const items = await findDynamicOptionTokensInFile(filePath);
+    result = result.concat(items);
+  }
+
+  return Array.from(new Set(result));
+}
+
+async function findDynamicOptionTokensInFile(filePath: string): Promise<string[]> {
```

Comment:
A couple things on this function I'm seeing:

1. It's returning results as "{{$token}}" instead of just "$token" which means you gotta clean it later.
2. It doesn't work with "{{ $token }}" or "{{ $token | transform }}"
3. I'm not sure we want a multiline search?

I'm not a regex expert but we want to match whatever is between 
1. {{ (optional whitespace)
2. (optional whitespace) (optional pipe |) (optional whitespace) (optional text) (optional whitespace) }}

I think that's possible with regex... I'll see if I can come up with something.

Reply:
Oops I got a bit mixed up with how handlebars uses transforms -- I was thinking in Angular pipe world... I will edit the comment above with the correct syntax.

Reply:
Here is a pull request that should change the regex to work as the comment above says: https://github.com/dj-doMains/blueprint/pull/2

I wanna double check whether you had the multiline search on purpose though?